libraries/AP_RSSI/AP_RSSI.cpp

```diff
@@ -153,7 +153,30 @@ float AP_RSSI::read_receiver_rssi()
     // should never get to here
     return 0.0f;
 }
-
+float AP_RSSI::read_receiver_link_quality()
+{
+    switch (RssiType(rssi_type.get())) {
+        case RssiType::TYPE_DISABLED:
+            return 0.0f;
+        case RssiType::ANALOG_PIN:
+            return read_pin_rssi();
+        case RssiType::RC_CHANNEL_VALUE:
+            return read_channel_rssi();
+        case RssiType::RECEIVER: {
+            int16_t quality = RC_Channels::get_receiver_link_quality();
+            if (quality != -1) {
+                return quality;
+            }
+            return 0.0f;
+        }
+        case RssiType::PWM_PIN:
+            return read_pwm_pin_rssi();
+        case RssiType::TELEMETRY_RADIO_RSSI:
+            return read_telemetry_radio_rssi();
+    }
+    // should never get to here
```

Comment:
I was being cautious to keep it at least show an rssi value if rssi type is other than receiver, at least it shows rssi as LQ too but I think its better we omit all the other cases and leave with just if(RssiType(rssi_type.get() == RssiType::RECEIVER)
{
            int16_t quality = RC_Channels::get_receiver_link_quality();
            if (quality != -1) {
                return quality;
            }
            return 0.0f;
        }'

Reply:
done